### PR TITLE
update CQL2 definition to reference OGC API yaml file in github, update npm build

### PR DESCRIPTION
**Pull request by philvarner**
**Related Issue(s):** 

- #241

**Proposed Changes:**

1. Update CQL2 spec to reference version in ogcapi-features github repo 
2. npm build improvements

**PR Checklist:**

- [X] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [X] This PR has **no** breaking changes.
- [X] This PR does not make any changes to the core spec in the `stac-spec` directory (these are included as a subtree and should be updated directly in [radiantearth/stac-spec](https://github.com/radiantearth/stac-spec))
- [X] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-api-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -35,7 +35,6 @@ build/Release
 # Dependency directories
 node_modules/
 jspm_packages/
-package-lock.json
```
Comment (philvarner) [inline .gitignore:38]: recommended practice is to commit package-lock.json

File: .nvmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+16
```
Comment (philvarner) [inline .nvmrc:1]: use nvmrc, and upgrade to node 16

File: collections/openapi.yaml [hunk 3]
```diff
@@ -12,14 +12,15 @@ info:
     name: Apache License 2.0
     url: 'http://www.apache.org/licenses/LICENSE-2.0'
 tags:
+  - name: Core
```
Comment (philvarner) [inline collections/openapi.yaml:15]: fixes a few warnings.

File: fragments/filter/README.md [hunk 4]
```diff
@@ -222,8 +222,6 @@ Formal definitions and grammars for CQL2 can be found here:
   - [cql.bnf](https://github.com/opengeospatial/ogcapi-features/blob/master/extensions/cql/standard/schema/cql.bnf)
   - [cql.json](https://github.com/opengeospatial/ogcapi-features/blob/master/extensions/cql/standard/schema/cql.json)
   - [cql.yml](https://github.com/opengeospatial/ogcapi-features/blob/master/extensions/cql/standard/schema/cql.yml)
-- A JSON Schema for only the parts of the CQL2 JSON encoding required by this extension is [here](cql.json)
```
Comment (philvarner) [inline fragments/filter/README.md:225]: edited cql.json has been removed, as we no longer have a custom version

**Review by cholmes — APPROVED** [review]
This looks good, but seems to also include some various clean-up updates in the config, pr template, gitignore, etc. Ideally those would be a separate PR. I don't think that's necessary, but maybe also mention those in the title of the PR?

File: .gitignore (hunk 1 above)
Comment (m-mohr) [reply .gitignore:38]: For software yes, but for a specification? There I'd assume you'd want the latest version?

File: .nvmrc (hunk 2 above)
Comment (m-mohr) [reply .nvmrc:1]: What's the reasoning? And why in a CQL PR?

Comment (philvarner) [reply .nvmrc:1]: I was updating the openapi yaml, so I took the opportunity to update the build which mostly does validation while I was at it.  I probably should have separated them into different PRs, but didn't.

**Pull request merged**